docs/src/developer/developer/upgrading.md

```diff
@@ -0,0 +1,32 @@
+
+- Your goal is to make all packages compile again with nix. Start small by trying to build single packages, e.g. `wire-api` or any external dependencies.
+```
+nix-build ./nix -A wireServer.haskellPackagesUnoptimizedNoDocs.wire-api
+```
+
+- When a dependency doesn't build anymore because of unmet version constraints see if you can use a never version (`haskell-pins.nix`, or `manual-overrides.nix` if multiple versions of the package are included in the `nixpkgs` pin). Also check out the git repo or issue tracker of the dependency to find forks that work. You can also try removing versions constraints of packages: see `doJailbreak` in `manual-overrides.nix`.
+
+- If test-suites of a dependency don't compile, or cannot be run because they require IO interaction you can disable them: see `dontCheck` in `manual-overrides.nix`
+
+- To force a rebuild of the nix shell delete the `.env` directory and run `direnv reload`
+
+- If you need to fix code in a dependency consider forking it and creating a PR to the upstream after successful adapation. Clone the repo and then symlink it inside `libs/` and run `generate-local-nix-packages.sh` to temporarily include its dependencies in the development shell. Make sure to include the package in `shellFor`. If you've got a working shell you can check the output of `ghc-pkg dump` to see the nix-provided ghc packages used by cabal.
+
+- Consider using `ghcid --command "cabal repl <package>"` when fixing packages
+
+- Delete the `dist-newstyle` directory everytime you upgrade the nix environment, without any exception. That is because cabal does not pick up changes to the `ghc-pkg` set (defined the development shell). Cabal will complain about missing dependencies in these cases.
+
+- When trying to build any packages with cabal (e.g. for fixing code in a depencency or fixing local packages), make sure you've got the right package set in the `shellFor` argument and the right transitive dependencies in `cabal.project`. It takes a couple of tries to get both: a nix provided environment that work and cabal not complaining about missing dependencies when building inside the environment.
+
+- It might happen that a package's test suite dependencies are not available in the nix environment. When you try to build with cabal it might try to build these external dependencies (which you want to avoid). What might work in these cases temporarily updating the `default.nix` file (generated by `generate-local-nix-packages.sh`) to add the test suits dependencies to the library section.
```

Comment:
```suggestion
- It might happen that a package's test suite dependencies are not available in the nix environment. When you try to build with cabal it might try to build these external dependencies (which you want to avoid). What might work in these cases is to temporarily update the `default.nix` file (generated by `generate-local-nix-packages.sh`) to add the test suits dependencies to the library section.
```